examples/server/server.cpp

```diff
@@ -2257,6 +2257,9 @@ struct server_context {
 
                 completion_token_output result;
                 const llama_token id = llama_sampling_sample(slot.ctx_sampling, ctx, NULL, slot.i_batch - i);
+                if (id == -1) {
+                    continue; // keep going, don't crash, already logged
```

Comment:
The `server` is still crashing when answering on the `/completion` endpoint with `bge-small`, but now with a different error.
```
GGML_ASSERT: llama.cpp:11006: seq_id < n_tokens && "seq_id cannot be larger than n_tokens with pooling_type == CLS"
```
At least it's no longer due to the logits. Or is it?

Some `fprintf` debugging right before the assertion:

```
llama_set_inputs: seq_id: 1, n_tokens: 5
llama_set_inputs: seq_id: 1, n_tokens: 5
llama_set_inputs: seq_id: 1, n_tokens: 5
llama_set_inputs: seq_id: 1, n_tokens: 5
llama_set_inputs: seq_id: 1, n_tokens: 5
llama_get_logits_ith_fail: invalid logits id 4, reason: no logits
llama_set_inputs: seq_id: 1, n_tokens: 1
GGML_ASSERT: llama.cpp:11007: seq_id < n_tokens && "seq_id cannot be larger than n_tokens with pooling_type == CLS"
```
Something is wrong in the iteration right after `llama_get_logits_ith` fails.

This seems to fix the problem:

```suggestion
                    send_error(slot, "can't get completions out of an embeddings model");
                    slot.cache_tokens.clear();
                    slot.reset();
                    slot.release();
                    slot.i_batch = -1;
                    continue; // continue loop of slots
```